--- phorum-mod-sphinx_search-1.0.0/sphinx_search.php	2010-04-14 16:28:33.609384236 +0300
+++ phorum-mod-sphinx_search-1.0.0/sphinx_search.php	2010-04-14 16:32:02.824588727 +0300
@@ -84,6 +84,15 @@
 
     // do the actual query
     $results = $sphinx->Query($arrSearch['search'],$index);
+
+    $res = $sphinx->GetLastWarning();
+    if ($res) {
+        error_log("sphinx_search.php: WARNING: $res");
+    }
+    $res = $sphinx->GetLastError();
+    if ($res) {
+        error_log("sphinx_search.php: ERROR: $res");
+    }
     
 	// if no messages were found, then return empty handed.
     if (! isset($results["matches"])) {
@@ -104,9 +113,6 @@
     krsort($found_messages);
     reset($found_messages);
     
-//    print_r($found_messages);/
-    
-	
     // prepare the array for building highlighted excerpts
     $docs=array();
     foreach($found_messages as $id => $data) {
@@ -123,7 +129,14 @@
     // build highlighted excerpts
     $highlighted = $sphinx->BuildExcerpts($docs,$excerpts_index,$words,$opts);
 
-    print $sphinx->GetLastError();
+    $res = $sphinx->GetLastWarning();
+    if ($res) {
+        error_log("sphinx_search.php: WARNING: $res");
+    }
+    $res = $sphinx->GetLastError();
+    if ($res) {
+        error_log("sphinx_search.php: ERROR: $res");
+    }
 
     $cnt=0;
     foreach($found_messages as $id => $content) {
