--- sphinx_search/sphinx_search.php~	2010-04-14 16:39:47.000000000 +0300
+++ sphinx_search/sphinx_search.php	2010-06-15 00:35:42.346275766 +0300
@@ -26,7 +26,7 @@
     // these are the index-names set in sphinx.conf - one for searching messages, the other for searching by authors only
     // both contain an additional index for the deltas - changes done after the last full reindex
     $index_name_msg = "phorum5_msg_d phorum5_msg";
-    $index_name_author = "phorum5_author phorum5_author_delta";
+    $index_name_author = "phorum5_author phorum5_author_d";
     
     // excerpts_index is just one index as that function only accepts one, it used for determining charsets / mapping tables, nothing more
     $excerpts_index = "phorum5_msg";  
--- phorum-mod-sphinx_search-1.0.0/README~	2010-06-15 00:43:33.000000000 +0300
+++ phorum-mod-sphinx_search-1.0.0/README	2010-06-15 00:44:08.319793287 +0300
@@ -158,7 +158,7 @@
                 WHERE message_id<=( SELECT max_doc_id FROM sph_counter WHERE counter_id=1 and type = 'author' ) and status=2
 }
 
-source phorum5_author_delta : phorum5_base
+source phorum5_author_d : phorum5_base
 {
 
         # main document fetch query
@@ -180,10 +180,10 @@
 
 
 
-index phorum5_author_delta : phorum5_base
+index phorum5_author_d : phorum5_base
 {
-        source = phorum5_author_delta
-        path = /usr/local/sphinx/var/data/phorum5_author_delta
+        source = phorum5_author_d
+        path = /usr/local/sphinx/var/data/phorum5_author_d
 }
 
 
--- phorum-mod-sphinx_search-1.0.0/sphinx.conf~	2010-04-14 16:38:35.000000000 +0300
+++ phorum-mod-sphinx_search-1.0.0/sphinx.conf	2010-06-15 00:44:28.003527305 +0300
@@ -116,7 +116,7 @@
 			WHERE message_id<=( SELECT max_doc_id FROM sph_counter WHERE counter_id=1 and type = 'author' ) and status=2
 }
 
-source phorum5_author_delta : phorum5_base
+source phorum5_author_d : phorum5_base
 {
 	# main document fetch query
 	# you can specify any number of fields
@@ -137,10 +137,10 @@
 }
 
 
-index phorum5_author_delta : phorum5_base
+index phorum5_author_d : phorum5_base
 {
-	source              = phorum5_author_delta
-	path                = /var/lib/sphinx/phorum5/phorum5_author_delta
+	source              = phorum5_author_d
+	path                = /var/lib/sphinx/phorum5/phorum5_author_d
 	charset_type        = utf-8
 }
 
