--- phorum-mod-sphinx_search-1.0.0/settings.php~	2010-03-30 17:45:42.000000000 +0300
+++ phorum-mod-sphinx_search-1.0.0/settings.php	2010-03-30 17:49:15.659963871 +0300
@@ -1,7 +1,7 @@
 <?php
     if (!defined("PHORUM_ADMIN")) return;
     
-	require_once("./mods/sphinx_search/defaults.php");
+	require_once 'defaults.php';
 	
     // save settings
     if (count($_POST))
@@ -28,7 +28,7 @@
     <br style="clear:both" />
     <?php
 
-    include_once "./include/admin/PhorumInputForm.php";
+    include_once PHORUM_INCLUDES_DIR.'/admin/PhorumInputForm.php';
     $frm = new PhorumInputForm ("", "post", "Save");
     $frm->hidden("module", "modsettings");
     $frm->hidden("mod", "sphinx_search");
--- phorum-mod-sphinx_search-1.0.0/sphinx_search.php~	2010-06-15 00:37:12.000000000 +0300
+++ phorum-mod-sphinx_search-1.0.0/sphinx_search.php	2010-06-15 00:38:37.226287129 +0300
@@ -15,13 +15,17 @@
 
 if (!defined("PHORUM")) return;
 
-require_once("./mods/sphinx_search/defaults.php");
+require_once 'defaults.php';
 
 function sphinx_search_action($arrSearch)
 {
 	global $PHORUM;
 
-    include './mods/sphinx_search/sphinxclient.php';
+	// No pecl class, try php version
+	if (!class_exists('SphinxClient')) {
+		// loads from php include_path
+		require_once 'sphinxapi.php';
+	}
 	
     // these are the index-names set in sphinx.conf - one for searching messages, the other for searching by authors only
     // both contain an additional index for the deltas - changes done after the last full reindex
@@ -147,6 +164,3 @@
     
 	return $arrSearch;
 }
-
-
-?>
\ No newline at end of file
