--- phorum-mod-sphinx_search-1.0.0/sphinx_search.php~	2010-06-15 13:36:55.000000000 +0300
+++ phorum-mod-sphinx_search-1.0.0/sphinx_search.php	2010-06-15 13:37:58.332229871 +0300
@@ -51,7 +51,8 @@
             return $arrSearch;
     }
     
-    $sphinx = new SphinxClient($PHORUM["mod_sphinx_search"]["hostname"], $PHORUM["mod_sphinx_search"]["port"]);
+    $sphinx = new SphinxClient();
+    $sphinx->SetServer($PHORUM["mod_sphinx_search"]["hostname"], $PHORUM["mod_sphinx_search"]["port"]);
     $sphinx->SetMatchMode($match_mode);
     
     // set the limits for paging
